src/python/CMakeLists.txt

```
@@ -1,28 +1,41 @@
 if (NOT STATIC_ONLY)
 
-set(PYTHON_HEADERS
-)
-
-set(PYTHON_SOURCES
-    bindings.cc
-)
+find_package(pybind11 CONFIG REQUIRED)
+find_package(Python 3.7 COMPONENTS Interpreter Development REQUIRED)
```

```suggestion
find_package(Python 3.7 COMPONENTS Interpreter Development NumPy REQUIRED)
```

also reset the top-level CMake min from 3.1 to 3.16. the `find_package(Python ...)` (note mixed-case Python) was new around 3.12 and not really stable until about 3.15.
https://github.com/jturney/ambit/pull/43/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR21

along with this change, you can find the FindPythonLibsNew.cmake file in the repo and delete it. That was the old way, borrowed from pb11 borrowed from numpy, to overcome cmake's bad old `find_package(PythonLibs)`